packages/integration_test/lib/_callback_io.dart

```diff
@@ -54,9 +59,43 @@ class IOCallbackManager implements CallbackManager {
     // comes up in the future. For example: `WebCallbackManager.cleanup`.
   }
 
+  // Whether the Flutter surface uses an Image.
+  bool _usesFlutterImage = false;
+
   @override
-  Future<void> takeScreenshot(String screenshot) {
-    throw UnimplementedError(
-        'Screenshots are not implemented on this platform');
+  Future<void> convertFlutterSurfaceToImage() async {
+    await integrationTestChannel.invokeMethod<void>(
+      'convertFlutterSurfaceToImage',
+      null,
+    );
+    _usesFlutterImage = true;
+  }
+
+  @override
+  Future<Map<String, dynamic>> takeScreenshot(String screenshot) async {
+    if (!_usesFlutterImage) {
+      throw 'Please call convertFlutterSurfaceToImage() before taking a screenshot';
```

Comment:
Throw a proper exception rather than a `String` - perhaps a `StateError` would be good here.

Also, avoid `Please` in the exception message (https://developers.google.com/style/tone#politeness)

Reply:
Done